Source/Core/DolphinQt/Main.cpp

```diff
@@ -265,6 +265,12 @@ int main(int argc, char* argv[])
 #if defined(USE_ANALYTICS) && USE_ANALYTICS
     if (!Config::Get(Config::MAIN_ANALYTICS_PERMISSION_ASKED))
     {
+      // To ensure that the analytics prompt appears aligned with the center of the main window,
+      // events are processed synchronously twice, allowing the main window to be shown and its
+      // layout to be calculated, as it is only on first show when its geometry is calculated.
+      QCoreApplication::processEvents();
+      QCoreApplication::processEvents();
```

Comment:
```suggestion
    QTimer::singleShot(0, &win, [win = &win] {
      if (Config::Get(Config::MAIN_ANALYTICS_PERMISSION_ASKED))
        return;
```
The `QCoreApplication::processEvents();` calls seem pretty hacky to me. What if you instead create the analytics window in a singleshot timer?

Reply:
> The `QCoreApplication::processEvents();` calls seem pretty hacky to me. What if you instead create the analytics window in a singleshot timer?

That can work only if the timeout is sufficiently great to give the main window time to show up. With a timeout of 0, the lambda will be called immediately from within the same thread, and therefore it suffers from the same issue.

With a slightly greater timeout (e.g. 20 ms), it should work, with the assumption that the main window will never take longer to show than the configured timeout. Do we prefer that over the synchronous event processing?

Another solution could be to define a dedicated timer and connect the slot with a `Qt::QueuedConnection` connection type. It should guarantee that the slot is called in the next event loop iteration, after the main window has been shown.

And another solution could be to move this action to `MainWindow::event()` (needs to be reimplemented). The first time that the `QEvent::Show` event is received, the analytics prompt would be shown.

Any thoughts or preference?

Reply:
I have updated it with an implementation that waits for the `Qt::ApplicationState::Active` state in the application.

It's different to what we had discussed offline, but it should be less involved than an event filter or implementing `MainWindow::event()`.